content/docs/reference-react-component.md

```diff
@@ -29,111 +29,112 @@ class Welcome extends React.Component {
+>React לא כופה על שימוש בתחביר מחלקות של גרסת ES6. אם אתם מעדיפים להימנע ממנו, אתם רשאיים להשתמש במודול `create-react-class` או הרחבה מותאמת דומה, במקום. העיפו מבט ב-[שימוש ב-React ללא גרסת ES6](/docs/react-without-es6.html) כדי ללמוד עוד.
 
-### The Component Lifecycle {#the-component-lifecycle}
+### קומפוננטת מעגל-החיים {#the-component-lifecycle}
 
-Each component has several "lifecycle methods" that you can override to run code at particular times in the process. **You can use [this lifecycle diagram](http://projects.wojtekmaj.pl/react-lifecycle-methods-diagram/) as a cheat sheet.** In the list below, commonly used lifecycle methods are marked as **bold**. The rest of them exist for relatively rare use cases.
+לכל קומפוננטה קיימות מספר "מתודות של מעגל חיים" שניתן לדרוס כדי להריץ קוד בזמנים מסוימים בתהליך. **אתם יכולים להשתמש [בדיאגרמת מעגל-החיים הזו](http://projects.wojtekmaj.pl/react-lifecycle-methods-diagram/) כשליף.** ברשימה מטה, מתודות מעגל-חיים שבשימוש נפוץ יותר, מסומנות **במודגש**. שאר המתודות קיימות עבור שימושים נדירים יותר.
 
-#### Mounting {#mounting}
+#### עיגון (Mounting) {#mounting}
 
-These methods are called in the following order when an instance of a component is being created and inserted into the DOM:
+קריאה למתודות הבאות מתבצעות בסדר הזה כאשר מופע של קומפוננטה נוצר ומוכנס לתוך ה-DOM:
 
 - [**`constructor()`**](#constructor)
 - [`static getDerivedStateFromProps()`](#static-getderivedstatefromprops)
 - [**`render()`**](#render)
 - [**`componentDidMount()`**](#componentdidmount)
 
->Note:
+>הערה:
 >
->These methods are considered legacy and you should [avoid them](/blog/2018/03/27/update-on-async-rendering.html) in new code:
->
->- [`UNSAFE_componentWillMount()`](#unsafe_componentwillmount)
+>המתודות האלו נחשבות למורשת וכדאי [להימנע מהן](/blog/2018/03/27/update-on-async-rendering.html) בקוד חדש:
+>- [`UNSAFE_componentWillUpdate()`](#unsafe_componentwillupdate)
+>- [`UNSAFE_componentWillReceiveProps()`](#unsafe_componentwillreceiveprops)
 
-#### Updating {#updating}
+#### עדכון {#updating}
 
-An update can be caused by changes to props or state. These methods are called in the following order when a component is being re-rendered:
+עדכון יכול להיגרם על ידי שינוי props או state. המתודות האלו נקראות בסדר להלן כאשר קומפוננטה מרונדרת מחדש:
 
 - [`static getDerivedStateFromProps()`](#static-getderivedstatefromprops)
 - [`shouldComponentUpdate()`](#shouldcomponentupdate)
 - [**`render()`**](#render)
 - [`getSnapshotBeforeUpdate()`](#getsnapshotbeforeupdate)
 - [**`componentDidUpdate()`**](#componentdidupdate)
 
->Note:
+
+>הערה:
 >
->These methods are considered legacy and you should [avoid them](/blog/2018/03/27/update-on-async-rendering.html) in new code:
+>המתודות להלן נחשבות "מורשת" (מתודות ישנות ולא מתוחזקות) ועדיף [להמנע משימוש בהן](/blog/2018/03/27/update-on-async-rendering.html) בקוד חדש:
```

Comment:
```suggestion
>המתודות להלן נחשבות ישנות ועדיף [להמנע משימוש בהן](/blog/2018/03/27/update-on-async-rendering.html) בקוד חדש:
```
Again, imo ישנות is a better translation for 'legacy'. @galtalmor what do you think?